fedimint-core/src/core.rs

```diff
@@ -110,11 +108,8 @@ macro_rules! module_dyn_newtype_impl_encode_decode {
                 r: &mut R,
                 modules: &$crate::module::registry::ModuleDecoderRegistry,
             ) -> Result<Self, DecodeError> {
-                $crate::core::encode::module_decode_key_prefixed_decodable(
-                    r,
-                    modules,
-                    |r, m, id| m.$decode_fn(r, id),
-                )
+                let key = ModuleInstanceId::consensus_decode(r, modules)?;
+                modules.get_expect(key).decode(r, key)
```

Comment:
nit: imho, variable names should have at least 3 letters. `item, err, ctx, ...`.  It makes things easier to read. Even for short methods like this; especially as methods may be extended in the future and suddenly a variable name is used in a long block of code. 
Feel free to ignore though, it's a bit unrelated to the changes.

Reply:
renamed `r` to `reader` (I just didn't change it previously)